session.h

```diff
@@ -14,13 +17,27 @@ class session
   session(boost::asio::io_service& io_service, std::map <std::string, std::string> function_mapping);
   tcp::socket& socket();
   void start();
-  int handle_write(const boost::system::error_code& error, size_t bytes_transferred);
-  void setup_obuffer(boost::asio::streambuf& out_streambuf, size_t bytes_transferred);
+  // sets up echo response object
+  int handle_echo(const boost::system::error_code& error, size_t bytes_transferred, http_response& response);
+  // sets up static response object
+  int handle_static(std::string& url, http_response& response);
+  // handles requests by seeing which type of response is required
   int handle_request(const boost::system::error_code& error, size_t bytes_transferred);
+  // writes out a string to the socket and closes the connection
+  void write_string(std::string send);
 
 private:
   std::vector<char> convert_buffer();
   std::string get_function_from_url(std::string url);
+  std::string get_path_from_url(std::string url);
+  // returns path to exec on current system
+  std::string get_exec_path();
+  // true if file exists
+  bool file_exists(std::string filename);
+  // returns content type header based on file extension
+  std::string get_content_type(std::string filename);
+  // reads raw file into vector of characters
+  std::vector<char> read_file(std::string filename);
   tcp::socket socket_;
```

Comment:
Basically what my previous comment was alluding to, have a separate class for handling echos and handling static file requests.